specutils/spectra/spectral_region.py

```diff
@@ -0,0 +1,152 @@
+import numpy as np
+import astropy.units as u
+
+
+class SpectralRegion:
+    """
+    A `SpectralRegion` is a container class enables some simplicty
+    in defining and passing a region (interval) for a spectrum.
+
+    In the future, there might be more functionality added in here and there
+    is some discussion that this might/could move to
+    `Astropy Regions <http://astropy-regions.readthedocs.io/en/latest/>`_.
+    """
+
+    @staticmethod
+    def from_center(center=None, width=None):
+        """
+        SpectralRegion class method that enables the definition of a `SpectralRegion`
+        from the center and width rather than lower and upper bounds.
+
+        Parameters
+        ----------
+
+        center: Scalar `~astropy.units.Quantity` with pixel or ``spectral_axis`` units
+           The center of the spectral region.
+
+        Width: Scalar `~astropy.units.Quantity` with pixel or ``spectral_axis`` units
+           The width of the spectral region.
+        """
+
+        return SpectralRegion(center - width, center + width)
+
+    def __init__(self, lower, upper):
+        """
+        Lower and upper values for the interval.
+
+        Parameters
+        ----------
+
+        lower: Scalar `~astropy.units.Quantity` with pixel or ``spectral_axis`` units
+           The lower bound of the region.
+
+        upper: Scalar `~astropy.units.Quantity` with pixel or ``spectral_axis`` units
+           The upper bound of the region.
+
+        """
+
+        # Create instance variables
+        self._lower = None
+        self._upper = None
+
+        # Set the values (using the setters for doing the proper checking)
+        self.lower = lower
+        self.upper = upper
+
+    @property
+    def lower(self):
+        return self._lower
+
+    @lower.setter
+    def lower(self, value):
+        """
+        Lower bound for the interval.
+
+        Parameters
+        ----------
+
+        value: Scalar `~astropy.units.Quantity` with pixel or ``spectral_axis`` units
```

Comment:
I may have used words like this before, but reading this now I think it might be confusing to users to say `` ` ` spectral_axis` ` units``. That is, they might not understand what that means exactly.  Maybe something like "any unit that is a valid unit for a ``spectral_axis``"?

Reply:
how about '... or spectral axis unit from `~astropy.units`' ?  (the suggested alternatively would make the line very long)

Reply:
Well, there's nothing in `astropy.units` that's specific to spectral axes... Maybe just ".. or any valid ``spectral_axis`` unit" ? That's only slightly longer,  (FWIW, this is a situation where I generally let the 80-char limit go by the wayside, since it's better to be specific enough that the user understands)